substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -0,0 +1,1527 @@
+			MultiViewImportNotificationSink::new_with_worker();
+
+		let mempool = Arc::from(TxMemPool::new(
+			pool_api.clone(),
+			listener.clone(),
+			Default::default(),
+			usize::MAX,
+		));
+
+		let (dropped_stream_controller, dropped_stream) =
+			MultiViewDroppedWatcherController::<ChainApi>::new();
+		let dropped_monitor_task = Self::dropped_monitor_task(dropped_stream, mempool.clone());
+
+		let combined_tasks = async move {
+			tokio::select! {
+				_ = import_notification_sink_task => {},
+				_ = dropped_monitor_task => {}
+			}
+		}
+		.boxed();
+
+		(
+			Self {
+				mempool,
+				api: pool_api.clone(),
+				view_store: Arc::new(ViewStore::new(pool_api, listener, dropped_stream_controller)),
+				ready_poll: Arc::from(Mutex::from(ReadyPoll::new())),
+				enactment_state: Arc::new(Mutex::new(EnactmentState::new(
+					best_block_hash,
+					finalized_hash,
+				))),
+				revalidation_queue: Arc::from(revalidation_worker::RevalidationQueue::new()),
+				import_notification_sink,
+				options: Options::default(),
+				is_validator: false.into(),
+				metrics: Default::default(),
+			},
+			combined_tasks,
+		)
+	}
+
+	/// Monitors the stream of dropped transactions and removes them from the mempool.
+	///
+	/// This asynchronous task continuously listens for dropped transaction notifications provided
+	/// within `dropped_stream` and ensures that these transactions are removed from the `mempool`
+	/// instance.
+	async fn dropped_monitor_task(
+		mut dropped_stream: StreamOfDropped<ChainApi>,
+		mempool: Arc<TxMemPool<ChainApi, Block>>,
+	) {
+		loop {
+			let Some(dropped) = dropped_stream.next().await else {
+				log::info!("fatp::dropped_monitor_task: terminated...");
+				break;
+			};
+			log::debug!("[{:?}] fatp::dropped notification, removing", dropped);
+			mempool.remove_dropped_transactions(&vec![dropped]).await;
+		}
+	}
+
+	/// Creates new fork aware transaction pool with the background revalidation worker.
+	///
+	/// The txpool essential tasks (including a revalidation worker) are spawned using provided
+	/// spawner.
+	pub fn new_with_background_worker(
+		options: Options,
+		is_validator: IsValidator,
+		pool_api: Arc<ChainApi>,
+		prometheus: Option<&PrometheusRegistry>,
+		spawner: impl SpawnEssentialNamed,
+		_best_block_number: NumberFor<Block>,
+		best_block_hash: Block::Hash,
+		finalized_hash: Block::Hash,
+	) -> Self {
+		let metrics = PrometheusMetrics::new(prometheus);
+		let listener = Arc::from(MultiViewListener::new());
+		let (revalidation_queue, revalidation_task) =
+			revalidation_worker::RevalidationQueue::new_with_worker();
+
+		let (import_notification_sink, import_notification_sink_task) =
+			MultiViewImportNotificationSink::new_with_worker();
+
+		let mempool = Arc::from(TxMemPool::new(
+			pool_api.clone(),
+			listener.clone(),
+			metrics.clone(),
+			TXMEMPOOL_TRANSACTION_LIMIT_MULTIPLIER * (options.ready.count + options.future.count),
```

Comment:
Views are not keeping the same set of transaction. It could happen that the view at block N already keeps max transactions, and we would have no space to keep some for block N+X.

So we need a little more space in internal mempool to accommodate for this.